app/src/main/java/com/android/fundamentals/workshop03/WS03ActorsFragment.kt

```diff
@@ -0,0 +1,71 @@
+package com.android.fundamentals.workshop03
+
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.RecyclerView
+import com.android.fundamentals.R
+import com.android.fundamentals.data.models.Actor
+import com.android.fundamentals.domain.ActorsDataSource
+import com.google.android.material.snackbar.Snackbar
+
+
+class WS03ActorsFragmentSolution : Fragment() {
+
+    private var recycler: RecyclerView? = null
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+
+        return inflater.inflate(R.layout.fragment_actors, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        recycler = view.findViewById(R.id.rv_actors)
+        //TODO 5: pass click listener to adapter
+        recycler?.adapter = WS03ActorsAdapter()
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        updateData()
+    }
+
+    override fun onDetach() {
+        recycler = null
+
+        super.onDetach()
+    }
+
+    private fun updateData() {
+        (recycler?.adapter as? WS03ActorsAdapter)?.apply {
+            bindActors(ActorsDataSource().getActors())
+            notifyDataSetChanged()
```

Comment:
Why notify is outside of the adapter? Isn't it functionality related to adapter?